src/main/webapp/resources/js/pages/password/PasswordReset.jsx

```diff
@@ -0,0 +1,242 @@
+import React from "react";
+import { render } from "react-dom";
+import { Alert, Button, Col, Form, Input, List, Row, Typography } from "antd";
+import {
+  EyeInvisibleOutlined,
+  EyeTwoTone,
+  LoadingOutlined,
+  LockOutlined,
+} from "@ant-design/icons";
+import { setBaseUrl } from "../../utilities/url-utilities";
+import { blue6 } from "../../styles/colors";
+import { SPACE_MD, SPACE_SM } from "../../styles/spacing";
+import { useSetPasswordMutation } from "../../apis/password-reset";
+import store from "../store";
+import { Provider } from "react-redux";
+
+const { Item } = Form;
+const passwordExpired =
+  new URLSearchParams(window.location.search).get("expired") || false;
+const passwordResetObj = window.PAGE.passwordReset;
+
+/**
+ * React component to render the Password Reset form.
+ * @returns {*}
+ * @constructor
+ */
+function PasswordResetForm() {
+  const [passwordResetForm] = Form.useForm();
+  const [setPassword] = useSetPasswordMutation();
+
+  const [loading, setLoading] = React.useState(false);
+  const [invalidPassword, setInvalidPassword] = React.useState(true);
+  const [updateSuccess, setUpdateSuccess] = React.useState(false);
+  const [updateError, setUpdateError] = React.useState(false);
+  const [errorMessages, setErrorMessages] = React.useState(null);
+  const passwordRef = React.useRef();
+
+  /**
+   * When the component gets added to the page,
+   * focus on the password input input.
+   */
+  React.useEffect(() => {
+    passwordRef.current.focus();
+    passwordRef.current.select();
+  }, []);
+
+  const passwordRules = [
+    i18n("PasswordReset.alert.rule2"),
+    i18n("PasswordReset.alert.rule3"),
+    i18n("PasswordReset.alert.rule4"),
+    i18n("PasswordReset.alert.rule5"),
+    i18n("PasswordReset.alert.rule6"),
+    i18n("PasswordReset.alert.recommendation1"),
+    i18n("PasswordReset.alert.recommendation2"),
+  ];
+
+  const submitPasswordResetForm = () => {
+    setLoading(true);
+    setPassword({
+      resetId: passwordResetObj.id,
+      password: passwordResetForm.getFieldValue("password"),
+    }).then((res) => {
+      if (res.error) {
+        setErrorMessages(res.error.data.errors);
+        setUpdateError(true);
+      } else {
+        setUpdateSuccess(true);
+        setUpdateError(false);
+      }
+      passwordResetForm.resetFields();
+      setLoading(false);
+    });
+  };
+
+  // Validator to check if password meets password policy requirements
+  const validatePassword = (password) => {
+    const passwordRegex =
+      /^(?=.*\d)(?=.*[!@#$%^&*()+?/<>={}.\\])(?=.*[a-z])(?=.*[A-Z]).{8,}$/;
+    const minimumPasswordLength = 8;
+    setInvalidPassword(true);
+
+    if (password.length !== 0) {
```

Comment:
Length should be handled by the input.  You only need custom ones here, I think there is even a regex one.

Reply:
Multiple messages are displayed if using the `min` rule for the input along with the validator. I think it's better to just have it as a part of our custom validator

![image](https://user-images.githubusercontent.com/25466211/170168089-682fc49a-6b9b-48ad-bfc2-fe43951633ae.png)

Reply:
**Currently works like this using just the custom validator:**

If less than 8 characters:

![image](https://user-images.githubusercontent.com/25466211/170176421-fcd76f1f-19b9-4430-940a-a95989ceb988.png)

If 8 characters or more but doesn't match the policy

![image](https://user-images.githubusercontent.com/25466211/170176474-a656ac3e-f4d6-45da-b485-c1d05a5a146d.png)

Reply:
Ok.  I see what you a getting at there.  Since you are listing the criteria above.  My only concern is that users will have been typing and then see that error and are like "WTF? what criteria am I not meeting"

Reply:
Leaving it as it is as discussed since the user can see the password policy and there is a sneak peek button on the input